subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -88,7 +89,12 @@ public class SigningExtension {
     /**
      * The provider of signatories.
      */
-    private SignatoryProvider signatories;
+    private SignatoryProvider<?> signatories;
+
+    /**
+     * The Sign tasks that have been created by this extension.
+     */
+    private final Map<String, Sign> signTasks;
```

Comment:
Have you considered using `tasks.getNames()` instead of creating a separate Map that literally duplicates the information?

Reply:
Thanks for spotting that out, this method seems to be in line with the lazy configuration. Will push a fix for this.